api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java

```diff
@@ -40,49 +43,71 @@ public class QaServingTranslator implements Translator<Input, Output> {
      */
     public QaServingTranslator(Translator<QAInput, String> translator) {
         this.translator = translator;
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public Batchifier getBatchifier() {
-        return translator.getBatchifier();
+        this.batchTranslator = translator.toBatchTranslator();
     }
 
     /** {@inheritDoc} */
     @Override
     public void prepare(TranslatorContext ctx) throws Exception {
         translator.prepare(ctx);
+        translator.prepare(ctx);
     }
 
     /** {@inheritDoc} */
     @Override
     public NDList processInput(TranslatorContext ctx, Input input) throws Exception {
         PairList<String, BytesSupplier> content = input.getContent();
+        if (content.isEmpty()) {
+            throw new TranslateException("Input data is empty.");
+        }
+
+        String contentType = input.getProperty("Content-Type", null);
         QAInput qa;
-        if (content.contains("question") && content.contains("paragraph")) {
-            String question = input.getAsString("question");
-            String paragraph = input.getAsString("paragraph");
-            qa = new QAInput(question, paragraph);
-        } else {
-            BytesSupplier data = input.getData();
-            if (data == null) {
-                throw new TranslateException("Input data is empty.");
-            }
+        if ("application/json".equals(contentType)) {
+            String json = input.getData().getAsString();
             try {
-                qa = JsonUtils.GSON.fromJson(data.getAsString(), QAInput.class);
+                JsonElement element = JsonUtils.GSON.fromJson(json, JsonElement.class);
+                if (element.isJsonArray()) {
+                    ctx.setAttachment("batch", Boolean.TRUE);
+                    QAInput[] inputs = JsonUtils.GSON.fromJson(json, QAInput[].class);
+                    return batchTranslator.processInput(ctx, inputs);
+                }
+
+                qa = JsonUtils.GSON.fromJson(json, QAInput.class);
             } catch (JsonParseException e) {
                 throw new TranslateException("Input is not a valid json.", e);
             }
+        } else if (content.contains("question") && content.contains("paragraph")) {
+            String question = input.getAsString("question");
+            String paragraph = input.getAsString("paragraph");
```

Comment:
Is "paragraph" the key we want to enforce here? It seems like in huggingface for QA they expect "context" https://huggingface.co/docs/transformers/tasks/question_answering#inference.

Reply:
We use paragraph for a long time, I made changes to compatible with huggingface

Reply:
Got it - I wasn't sure if it was compatible